pkg/agent/secondarynetwork/podwatch/controller.go

```diff
@@ -0,0 +1,373 @@
+        podSriovVFDeviceID, err := cniserver.GetPodContainerDeviceIDs(pod.Name, pod.Namespace)
+        if err != nil {
+               klog.ErrorS(err, "handleAddUpdatePod: GetPodContainerDeviceIDs failed for ", pod.Name, pod.Namespace)
+               // return nil here. we do not want to retry/requeue work if VF id fetch fails (user to check if VFs are created at node).
+               return err
+        }
+
+        if err = pc.cniServer.GetPodConfigurator().ConfigureSriovSecondaryInterface(
+                                      containerInfo.PodName,
+                                      containerInfo.PodNameSpace,
+                                      containerInfo.ContainerID,
+                                      containerInfo.ContainerNetNS,
+                                      netinfo.InterfaceName,
+                                      containerInfo.Mtu,
+                                      podSriovVFDeviceID[0],
+                                      result,
+                                      ); err != nil {
+                 klog.ErrorS(err, "Failed to configure interfaces for container ", containerInfo.ContainerID, err)
+                 return err
+        }
+	return nil
+}
+
+func (pc *PodController) configureVethPairAsSecondaryInterface(pod *corev1.Pod, netinfo *SecondaryNetworkObject, containerInfo *cnipodcache.CNIConfigInfo, result *current.Result) error {
+     return nil
+}
+
+func (pc *PodController) configureSecondaryInterface(pod *corev1.Pod, netinfo *SecondaryNetworkObject, podCNIInfo []*cnipodcache.CNIConfigInfo, cniconfig []byte) error {
+	var err error
+        var ipamResult *current.Result
+        var cmdArgs *invoke.Args
+
+        for _, containerInfo := range podCNIInfo {
+               cmdArgs = &invoke.Args{Command: string("ADD"), ContainerID: containerInfo.ContainerID,
+                       NetNS: containerInfo.ContainerNetNS, IfName: containerInfo.ContainerIfDev,
+                       Path: cniPath}
+
+               ipamResult, err = ipam.GetIPAMSubnetAddress(cniconfig, cmdArgs)
+               if err != nil {
+                      return errors.New("Secondary network global IPAM failed")
+               }
+
+               result := &current.Result{CNIVersion: containerInfo.CNIVersion}
+               result.IPs = ipamResult.IPs
+               result.Routes = ipamResult.Routes
+	       // Set result.Interface to container interface.
+               for _, ip := range result.IPs {
+                      ip.Interface = current.Int(1)
+               }
+               // If secondary interface name not provided in pod annotation, assign default name.
+               if len(netinfo.InterfaceName) == 0 {
+                    netinfo.InterfaceName = netDev
+               }
+
+               if netinfo.InterfaceType == sriov {
+		    pc.configureSriovAsSecondaryInterface(pod, netinfo, containerInfo, result)
+               } else {
```

Comment:
I suggest to check this earlier (e.g. at line 243, at least before IPAM), and log an error. We can add configureVethPairAsSecondaryInterface() later.